{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "envoy",
                    "vendor": "envoyproxy",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "< 1.18.6"
                        },
                        {
                            "status": "affected",
                            "version": ">= 1.19.0, < 1.19.3"
                        },
                        {
                            "status": "affected",
                            "version": ">= 1.20.0, < 1.20.2"
                        },
                        {
                            "status": "affected",
                            "version": ">= 1.21.0, < 1.21.1"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "Envoy is an open source edge and service proxy, designed for cloud-native applications. In affected versions of Envoy a crash occurs when configured for :ref:`upstream tunneling <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.tunneling_config>` and the downstream connection disconnects while the the upstream connection or http/2 stream is still being established. There are no workarounds for this issue. Users are advised to upgrade."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    }
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-416",
                            "description": "CWE-416: Use After Free",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2022-02-22T22:45:22",
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/envoyproxy/envoy/security/advisories/GHSA-cmx3-fvgf-83mf"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/envoyproxy/envoy/commit/ce0ae309057a216aba031aff81c445c90c6ef145"
                }
            ],
            "source": {
                "advisory": "GHSA-cmx3-fvgf-83mf",
                "discovery": "UNKNOWN"
            },
            "title": "Crash when tunneling TCP over HTTP in Envoy",
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "security-advisories@github.com",
                    "ID": "CVE-2021-43826",
                    "STATE": "PUBLIC",
                    "TITLE": "Crash when tunneling TCP over HTTP in Envoy"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "envoy",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "< 1.18.6"
                                                    },
                                                    {
                                                        "version_value": ">= 1.19.0, < 1.19.3"
                                                    },
                                                    {
                                                        "version_value": ">= 1.20.0, < 1.20.2"
                                                    },
                                                    {
                                                        "version_value": ">= 1.21.0, < 1.21.1"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "envoyproxy"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "Envoy is an open source edge and service proxy, designed for cloud-native applications. In affected versions of Envoy a crash occurs when configured for :ref:`upstream tunneling <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.tunneling_config>` and the downstream connection disconnects while the the upstream connection or http/2 stream is still being established. There are no workarounds for this issue. Users are advised to upgrade."
                        }
                    ]
                },
                "impact": {
                    "cvss": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    }
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "CWE-416: Use After Free"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/envoyproxy/envoy/security/advisories/GHSA-cmx3-fvgf-83mf",
                            "refsource": "CONFIRM",
                            "url": "https://github.com/envoyproxy/envoy/security/advisories/GHSA-cmx3-fvgf-83mf"
                        },
                        {
                            "name": "https://github.com/envoyproxy/envoy/commit/ce0ae309057a216aba031aff81c445c90c6ef145",
                            "refsource": "MISC",
                            "url": "https://github.com/envoyproxy/envoy/commit/ce0ae309057a216aba031aff81c445c90c6ef145"
                        }
                    ]
                },
                "source": {
                    "advisory": "GHSA-cmx3-fvgf-83mf",
                    "discovery": "UNKNOWN"
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "assignerShortName": "GitHub_M",
        "cveId": "CVE-2021-43826",
        "datePublished": "2022-02-22T22:45:22",
        "dateReserved": "2021-11-16T00:00:00",
        "dateUpdated": "2022-02-22T22:45:22",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}